src/dialogs/newDatabaseDialog.ts

```diff
@@ -0,0 +1,256 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as azdata from 'azdata';
+import { SqlOpsDataClient } from 'dataprotocol-client';
+import * as vscode from 'vscode';
+import { CancelButtonLabel, CreateButtonLabel, DatabaseCharsetDropDownLabel, DatabaseCharsetLabel, DatabaseCollationDropDownLabel, DatabaseCollationLabel, DatabaseNameLabel, DatabaseNameTextBoxLabel, DatabaseNameTextBoxPlaceHolder, NewDatabaseDetailsTitle, NewDatabaseDialogName, NewDatabaseTitle } from '../uiConstants';
+import { Deferred } from '../utils/PromiseUtils';
+import { ToolsServiceUtils } from '../utils/toolsserviceUtils';
+
+export class NewDatabaseDialog {
+	public dialog: azdata.window.Dialog;
+	public newDatabaseTab: azdata.window.DialogTab;
+	public databaseNameTextBox: azdata.InputBoxComponent | undefined;
+	public databaseCharsetDropDown: azdata.DropDownComponent | undefined;
+	public databaseCollationDropDown: azdata.DropDownComponent | undefined;
+	private connectionProfile: azdata.IConnectionProfile;
+	private connectionOwnerUri: string;
+	private client: SqlOpsDataClient;
+	private formBuilder: azdata.FormBuilder | undefined;
+	private toDispose: vscode.Disposable[] = [];
+	private initDialogComplete: Deferred = new Deferred();
+	private DEFAULT_CHARSET = "Default Charset"
+	private DEFAULT_COLLATION = "Default Collation"
+	private CHARSET_QUERY = "SELECT CHARACTER_SET_NAME FROM INFORMATION_SCHEMA.CHARACTER_SETS"
+	private COLLATION_QUERY = "SELECT COLLATION_NAME FROM INFORMATION_SCHEMA.COLLATIONS WHERE CHARACTER_SET_NAME="
+	private charsetValuesCache: string[] = [this.DEFAULT_CHARSET];
+
+	constructor(private profile: azdata.IConnectionProfile, client: SqlOpsDataClient) {
+		this.connectionProfile = profile;
+		this.client = client;
+		this.dialog = azdata.window.createModelViewDialog(NewDatabaseTitle, NewDatabaseDialogName);
+		this.newDatabaseTab = azdata.window.createTab(NewDatabaseTitle);
```

Comment:
the parameter value can be empty instead of `NewDatabaseTitle`

Reply:
done

Reply:
you can use ''.

Reply:
done